src/global/registry.rs

```diff
@@ -37,57 +41,94 @@ impl Registry {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct RegistryEntry {
-    flags: u16,
-    mime_type: u16,
-    content_version: u8,
-    signature: u32,
+    pub(crate) flags: u8,
+    pub(crate) content_version: u16,
+    pub(crate) blob_signature: [u8; ed25519_dalek::SIGNATURE_LENGTH],
+    
+    pub(crate) path_name_length: u16,
+    pub(crate) path: Vec<u8>, // length of path_name_length
+    
+    pub(crate) compressed_size: u32,
+    pub(crate) uncompressed_size: u32,
 
-    path_name_start: RegisterType,
-    path_name_end: RegisterType,
-
-    index: RegisterType,
-    offset: RegisterType,
+    pub(crate) byte_offset: u64, // offset of the blob from the beginning of the file
+    
+    pub(crate) mime_type: u16,
 }
 
 impl RegistryEntry {
-    pub const SIZE: usize = 41;
-    pub fn from_bytes(buffer: [u8; Self::SIZE]) -> anyhow::Result<RegistryEntry> {
-        Ok(RegistryEntry{
-            flags: u16::from_ne_bytes(buffer[0..2].try_into()?),
-            mime_type: u16::from_ne_bytes(buffer[2..4].try_into()?),
-            content_version: *buffer.get(4).ok_or(anyhow::Error::msg("Out of bounds error"))?,
-            signature: u32::from_ne_bytes(buffer[5..9].try_into()?),
-            path_name_start: RegisterType::from_ne_bytes(buffer[9..17].try_into()?),
-            path_name_end: RegisterType::from_ne_bytes( buffer[17..25].try_into()?),
-            index: RegisterType::from_ne_bytes(buffer[25..33].try_into()?),
-            offset: RegisterType::from_ne_bytes(buffer[33..41].try_into()?)
-        })
+    /// size in bytes without the path (since it is variable-length)
```

Comment:
The registry entry is stored in the registry. All registry entries have the same size, otherwise, we would need null terminators to mark where an entry ends. No need for that when we already have a solid way of storing large boundless data. A sort of pointer, the way vectors work in rust. So entries naturally cannot store the pathname. Only a reference to where it is stored in the glob. _Pathnames are a facade_. Entries are only pointers.

Reply:
But if we want to mount the archive into a virtual filesystem, we need to know the paths. So isn't it more efficient to store the path in the header instead of seeking around in the file for every path?